python/pyspark/sql/pandas/typehints.py

```diff
@@ -138,9 +138,8 @@ def infer_eval_type(
 def check_tuple_annotation(
     annotation: Any, parameter_check_func: Optional[Callable[[Any], bool]] = None
 ) -> bool:
-    # Python 3.6 has `__name__`. Python 3.7 and 3.8 have `_name`.
```

Comment:
Ah, we can just fix the comments here like: `Tuple` has `_name` but other types have `__name__`, and recoverthe codes `, getattr(annotation, "__name__", None)`.

Here can take other instances than `Tuple` that have `__name__` attributes. I should have explained that in the comment here ..